packages/block-editor/src/components/block-edit/edit.js

```diff
@@ -48,27 +57,223 @@ const Edit = ( props ) => {
 const EditWithFilters = withFilters( 'editor.BlockEdit' )( Edit );
 
 const EditWithGeneratedProps = ( props ) => {
-	const { attributes = {}, name } = props;
+	const { name, clientId, attributes, setAttributes } = props;
```

Comment:
In the previous declaration, `attributes` had a default empty object. Could this cause any issues if it is undefined? Maybe some developers are relying on it being an object in `BlockEdit`?

Reply:
I don't think there was ever guaranteed that `attributes` is going to be an object due to historical reasons. `computedAttributes` gets passed down instead, which is always an object. I'm not entirely sure why it would get mapped to an empty object in this place, but if we want to keep that, then we should use a stable reference instead of overriding it with a new empty object instance on every re-render.

Reply:
On the server, attributes can be set to `null`:
https://github.com/WordPress/wordpress-develop/blob/32880ec4e910837cd267e1685733867b243e22fa/src/wp-includes/class-wp-block-type.php#L169-L175

The same applies to REST API response:

https://github.com/WordPress/wordpress-develop/blob/32880ec4e910837cd267e1685733867b243e22fa/src/wp-includes/rest-api/endpoints/class-wp-rest-block-types-controller.php#L503-L513

It was hardened in `block.json` schema:

https://github.com/WordPress/gutenberg/blob/af34d1ef0bf3db2913c71fb1e9686c34a61f3ee2/schemas/json/block.json#L107-L110

However, when not provided on the server, that would still mean it's `null`.

On the client, it defaults to an empty object:

https://github.com/WordPress/gutenberg/blob/af34d1ef0bf3db2913c71fb1e9686c34a61f3ee2/packages/blocks/src/store/process-block-type.js#L235-L239

However, I'm not entirely sure what happens if the server sets `null` when bootstraping the value.

Reply:
I think for the block type processing, `attributes` will never be `null` because it defines some attributes that are common to all blocks: `metadata` and `lock`: https://github.com/WordPress/wordpress-develop/blob/32880ec4e910837cd267e1685733867b243e22fa/src/wp-includes/class-wp-block-type.php#L282-L285.

If `null` is a valid value for the `BlockEdit`, I'm fine keeping it this way.

Reply:
Good point, it's probably impossible to see `attributes` set to `null` on the client based on the fact it's always contains global attributes on the server:

https://github.com/WordPress/wordpress-develop/blob/32880ec4e910837cd267e1685733867b243e22fa/src/wp-includes/class-wp-block-type.php#L553-L563